src/test/java/com/google/sps/SentimentServletTest.java

```diff
@@ -23,48 +28,71 @@
   @Mock
   HttpServletRequest requestMock;
 
+  @Spy
+  HttpServletResponse responseMock;
+
   /**
-   * Takes a Json String of an array and returns the elements of the array as a String, separated
-   * by periods.
-   * Ex. "["foo","bar"]" -> "foo. bar"
+   * A Json String in the form of an array should be converted to a String of period separated
+   * comments. A sentiment score should be calculated and the score should be printed in a Json
+   * object. 
+   * Ex. "foo" -> "{"commentScore":X.XX}" where X.XX is a float, for any # of decimal places. 
    */
   @Test
-  public void jsonArrayConvertsToPeriodSeparatedString() throws IOException {
+  public void jsonArrayPrintsVideoAnalysisJsonObject() throws IOException {
     String json = "[\"a\",\"b\",\"c\"]"; // "["a","b","c"]"
     when(requestMock.getReader()).thenReturn(new BufferedReader(new StringReader(json)));
     
-    String actual = sentimentServlet.requestToString(requestMock);
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    when(responseMock.getWriter()).thenReturn(writer);
+
+    new SentimentServlet().doGet(requestMock, responseMock);
 
-    Assert.assertEquals("a. b. c", actual);
+    String pattern = "\\{\"commentScore\":[-+]?[0-9]*\\.?[0-9]+\\}\n";
+
+    Assert.assertTrue(stringWriter.toString().matches(pattern));
   }
 
   /**
-   * If the Json String is an empty array, an empty String should be returned. 
-   * Ex. "[]" -> ""
+   * If the Json String is not in the form of an array, a JsonSyntaxException is caught and null is
+   * returned. Gson excludes null values, therefore resulting in an empty Json object.
+   * Ex. "foo" -> "{}"
    */
   @Test
-  public void emptyJsonArrayConvertsToEmptyString() throws IOException {
-    String json = "[]";
+  public void notArrayPrintsEmptyJsonObject() throws IOException {
```

Comment:
change this test after addressing the comment above to assert if an error was return in the HTTP response.